.github/workflows/zosEndToEndTest.yml

```diff
@@ -0,0 +1,63 @@
+name: z/OS end-to-end Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+
+  run-analysis-on-zos:
+    permissions:
+      id-token: write
+    name: Run analysis on zOS
+    runs-on: sonar-s-public
+    env:
+      ZOS_PRIVATE_KEY: "zos_key"
```

Comment:
Is it really used?

Reply:
What do you mean? Is there another way to do it? For example I came across [this stackoverflow](https://stackoverflow.com/questions/63012327/ssh-host-key-verification-failed-in-github-actions-but-key-exists-in-known-h). Would it be a solution and then I would not have to pass the ssh key everywhere?

Reply:
Isn't the variable defined by the Vault step, line 26?

Reply:
That would be the content of the key which I get from Vault (It just has the same name). Then I write the content in a new file located by this env variable